File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        n = len(nums)
        
        maxSum = nums[0];
        currSum = nums[0];
        
        for i in range(1, n):
            sum = currSum + nums[i];
            if sum < nums[i]:
                currSum = nums[i];
            else:
                currSum = sum;
            
            maxSum = max(maxSum, currSum);
        
        return maxSum;

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int n = nums.size();
        
        int maxSum = nums[0];
        int currSum = nums[0];
        for (int i = 1; i < n; i++) {
            int sum = currSum + nums[i];
            if (sum < nums[i]) {
                currSum = nums[i];
            } else {
                currSum = sum;
            }
            
            maxSum = max(maxSum, currSum);
        }
        
        return maxSum;
    }
};